checker/bin/wpi.sh

```diff
@@ -291,22 +294,26 @@ elif [ "${has_java16}" = "yes" ]; then
 fi
 configure_and_exec_dljc "$@"
 
-if [ "${has_java11}" = "yes" ] && [ "${WPI_RESULTS_AVAILABLE}" != "yes" ]; then
-    # If running under Java 11 fails, try Java 8.
-    if [ "${has_java8}" = "yes" ]; then
-      export JAVA_HOME="${JAVA8_HOME}"
-      echo "couldn't build using Java 11; trying Java 8"
-      configure_and_exec_dljc "$@"
-    fi
+# If results aren't available after the first run, then re-run with Java 11 if
+# it is available and the first run used Java 8 (since Java 8 has the highest priority,
+# the first run using Java 8 is equivalent to Java 8 being available).
+if [ "${WPI_RESULTS_AVAILABLE}" != "yes" ] && [ "${has_java11}" = "yes" ]; then
+  if [ "${has_java8}" = "yes" ]; then
+    export JAVA_HOME="${JAVA11_HOME}"
+    echo "couldn't build using Java 8; trying Java 11"
+    configure_and_exec_dljc "$@"
+  fi
 fi
 
-if [ "${has_java11}" = "yes" ] && [ "${has_java8}" = "yes" ] && [ "${WPI_RESULTS_AVAILABLE}" != "yes" ]; then
-    # If running under Java 11 and Java 8 fails, try Java 16.
-    if [ "${has_java16}" = "yes" ]; then
-      export JAVA_HOME="${JAVA16_HOME}"
-      echo "couldn't build using Java 11 or Java 8; trying Java 16"
-      configure_and_exec_dljc "$@"
-    fi
+# If results still aren't available and Java 16 is available, then Java 16 has not yet
```

Comment:
The text of this comment block and the one above has gotten out of sync.

Reply:
They did! Sorry about that. I've fixed them both to be more like the first one.